src/xunit.performance.api/Model/AssemblyModel.cs

```diff
@@ -175,6 +176,118 @@ private static IEnumerable<(string testCaseName, string metric, IEnumerable<doub
+        public string Namespace { get; set; }
+        
+        [XmlArray("Tests")]
+        public List<ScenarioTestModel> Tests { get; set; }
+
+        public ScenarioBenchmark()
+        {
+            Tests = new List<ScenarioTestModel>();
+            Namespace = "";
+        }
+
+        public ScenarioBenchmark(string name)
+        {
+            Name = name;
+            Namespace = "";
+            Tests = new List<ScenarioTestModel>();
+        }
+
+        public void Serialize(string xmlFileName)
+        {
+            var namespaces = new XmlSerializerNamespaces();
+            namespaces.Add("", "");
+            using (var stream = File.Create(xmlFileName))
+            {
+                using (var sw = new StreamWriter(stream))
+                {
+                    new XmlSerializer(typeof(ScenarioBenchmark))
+                        .Serialize(sw, this, namespaces);
+                }
+            }
+        }
+
+        internal DataTable GetStatistics()
+        {
+            //fixme. Column names should be defined as constant methods
+            var dt = new DataTable();
+            var col0_testName = dt.AddColumn("Test Name");
+            var col1_metric = dt.AddColumn("Metric");
+            var col2_iterations = dt.AddColumn("Iterations");
+            var col3_average = dt.AddColumn("AVERAGE");
+            var col4_stdevs = dt.AddColumn("STDEV.S");
+            var col5_min = dt.AddColumn("MIN");
+            var col6_max = dt.AddColumn("MAX");
+
+            foreach (var test in Tests)
+            {
+                foreach (var metric in test.Performance.Metrics)
+                {
+                    var values = test.Performance.IterationModels
+                        .Where(iter => iter.Iteration.ContainsKey(metric.Name))
+                        .Select(iter => iter.Iteration[metric.Name]);
+
+                    var count = values.Count();
+                    if (count == 0) // Cannot compute statistics when there are not results (e.g. user only ran a subset of all tests).
+                        continue;
+
+                    var avg = values.Average();
+                    var stdev_s = Math.Sqrt(values.Sum(x => Math.Pow(x - avg, 2)) / (count - 1));
+                    var max = values.Max();
+                    var min = values.Min();
+
+                    var newRow = dt.AppendRow();
+                    newRow[col0_testName] = test.Name;
+                    newRow[col1_metric] = metric.DisplayName;
+
+                    newRow[col2_iterations] = count.ToString();
+                    newRow[col3_average] = avg.ToString();
+                    newRow[col4_stdevs] = stdev_s.ToString();
+                    newRow[col5_min] = min.ToString();
+                    newRow[col6_max] = max.ToString();
+                }
+            }
+
+            return dt;
+        }
+    }
+
+    [Serializable]
+    [XmlType("Test")]
+    public sealed class ScenarioTestModel
```

Comment:
I do believe we could do something less constraining, but I'm not sure how much different the format might look at the end, so in the end I wonder whether the value of iterating on the design further is worth too much. Happy to discuss ideas offline though.